src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -33,17 +33,15 @@ public class ModuleIntrinsics
         [TraceSource("Modules", "Module loading and analysis")]
```

Comment:
@iSazonov, your last commit had 6 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

TabCompletion.Should complete abbreviated function
```powershell
Expected exactly 'PSConsoleHostReadLine', but got @('PSConsoleHostReadLine', 'PSReadLine\PSConsoleHostReadLine').
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1: line 22
22:         $res | Should -BeExactly 'PSConsoleHostReadLine'
```
Import-Module.should be able to add a module with using ModuleInfo switch
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 26
26:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```
Import-Module.should be able to load a module with a trailing directory separator: @('/opt/microsoft/powershell/6/Modules/Microsoft.PowerShell.Security', '/home/vsts/work/1/a/bins/publish/Modules/Microsoft.PowerShell.Security', /)
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 26
26:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```
DSC MOF Compilation.Should be able to compile a MOF from a basic configuration
```powershell
Exception calling "Create" with "1" argument(s): "At line:1 char:23
+         configuration DSCTestConfig
+                       ~~~~~~~~~~~~~
System.IO.DirectoryNotFoundException: Unable to find DSC schema store at /etc/opt/omi/conf/dsc/configuration. Please ensure PS DSC for Linux is installed.
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.Initialize(Collection`1 errors, List`1 modulePathList)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Dictionary`2 functionsToDefine, Collection`1 errors, List`1 modulePathList, Boolean cacheResourcesFromMultipleModuleVersions)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Collection`1 errors)
   at System.Management.Automation.Language.Parser.ConfigurationStatementRule(IEnumerable`1 customAttributes, Token configurationToken)

At line:2 char:9
+         {
+         ~
Unexpected token '{' in expression or statement.

At line:10 char:9
+         }
+         ~
Unexpected token '}' in expression or statement."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PSDesiredStateConfiguration/MOF-Compilation.Tests.ps1: line 24
24:         [Scriptblock]::Create(@"

```
DSC MOF Compilation.Should be able to compile a MOF from another basic configuration
```powershell
Exception calling "Create" with "1" argument(s): "At line:1 char:23
+         configuration DSCTestConfig
+                       ~~~~~~~~~~~~~
System.IO.DirectoryNotFoundException: Unable to find DSC schema store at /etc/opt/omi/conf/dsc/configuration. Please ensure PS DSC for Linux is installed.
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.Initialize(Collection`1 errors, List`1 modulePathList)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Dictionary`2 functionsToDefine, Collection`1 errors, List`1 modulePathList, Boolean cacheResourcesFromMultipleModuleVersions)
   at Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(Collection`1 errors)
   at System.Management.Automation.Language.Parser.ConfigurationStatementRule(IEnumerable`1 customAttributes, Token configurationToken)

At line:2 char:9
+         {
+         ~
Unexpected token '{' in expression or statement.

At line:13 char:9
+         }
+         ~
Unexpected token '}' in expression or statement."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PSDesiredStateConfiguration/MOF-Compilation.Tests.ps1: line 43
43:         [Scriptblock]::Create(@"

```